code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm

```diff
@@ -343,25 +343,33 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 
 	patient.add_filter("hivelord_healing_infusion_outline", 3, list("type" = "outline", "size" = 1, "color" = COLOR_VERY_PALE_LIME_GREEN)) //Set our cool aura; also confirmation we have the buff
 
+	var/datum/healing_infusion_storage_datum/storage_datum = new /datum/healing_infusion_storage_datum //Create the buff datum
+
 	patient.infusion_active = TRUE //Indicate the infusion as being active
 
-	health_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
-	sunder_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
+	storage_datum.health_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
+	storage_datum.sunder_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
 
-	addtimer(CALLBACK(src, .proc/healing_infusion_timer_check, patient), HIVELORD_HEALING_INFUSION_DURATION)
+	addtimer(CALLBACK(storage_datum, /datum/healing_infusion_storage_datum/.proc/healing_infusion_timer_check, patient), HIVELORD_HEALING_INFUSION_DURATION)
```

Comment:
this datum is bad and you should use the upgrade signal to cancel it instead

Reply:
Thanks to my other PR, this is probably no longer necessary. I can change this to a balance PR however